src/main/java/com/ldtteam/blockout/views/Tab.java

```diff
@@ -0,0 +1,71 @@
+package com.ldtteam.blockout.views;
+
+import com.ldtteam.blockout.Pane;
+import com.ldtteam.blockout.PaneBuilders;
+import com.ldtteam.blockout.PaneParams;
+import com.ldtteam.blockout.Parsers;
+import com.ldtteam.blockout.controls.Text;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.text.IFormattableTextComponent;
+import net.minecraft.util.text.ITextComponent;
+import net.minecraft.util.text.StringTextComponent;
+
+/**
+ * A form of view to be used only in a {@link TabSet}
+ * and typically associated with a {@link TabSet.ButtonTab} used to select it.
+ */
+public class Tab extends View
+{
+    /** The icon to be shown on any corresponding tab button */
+    protected ResourceLocation icon;
+
+    /** The text to use as a tooltip, or as the label itself when icon is null */
+    protected IFormattableTextComponent label;
+
+    /**
+     * Constructs a new tab from the given information.
+     * NOTE: this will not add it to a tab set, for that use {@link TabSet#addTab(Tab)}
+     * @param title the label for the tooltip
+     * @param icon a resource string to the icon relative to the gui textures folder
+     * @param content each element to be included in this view
+     */
+    public Tab(String title, String icon, Pane... content)
```

Comment:
suggestion: might want to hide that and use ctor directly in TabSet#addChild()

Reply:
Not quite sure what you mean but this should be used for dynamic element creation by planned use cases. It's fine as it is, and I removed the vararg.